code/modules/reagents/chemistry/taste.dm

```diff
@@ -0,0 +1,44 @@
+
+//================================TASTE===================================================
+/**
+ * Returns what this reagents in our given list taste like
+ *
+ * Arguments:
+ * * list/reagent_list - List of reagents to taste.
+ * * mob/living/taster - Who is doing the tasting. Some mobs can pick up specific flavours.
+ * * minimum_percent - The lower the minimum percent, the more sensitive the message is.
+ * * weight_modifier - Value to multiply each reagent's taste weight with.
+ */
+/proc/generate_reagents_taste_message(list/reagent_list, mob/living/taster, minimum_percent)
+	var/list/out = list()
+	var/list/tastes = list() //descriptor = strength
+	if(minimum_percent <= 100)
+		for(var/datum/reagent/reagent as anything in reagent_list)
+			if(!reagent.taste_mult)
+				continue
+
+			var/list/taste_data = reagent.get_taste_description(taster)
+			for(var/taste in taste_data)
+				if(taste in tastes)
+					tastes[taste] += taste_data[taste] * reagent.volume * reagent.taste_mult
+				else
+					tastes[taste] = taste_data[taste] * reagent.volume * reagent.taste_mult
+		//deal with percentages
+		// TODO it would be great if we could sort these from strong to weak
+		var/total_taste = counterlist_sum(tastes)
+		if(total_taste > 0)
+			for(var/taste_desc in tastes)
+				var/percent = tastes[taste_desc]/total_taste * 100
+				if(percent < minimum_percent)
+					continue
+				var/intensity_desc = "a hint of"
+				if(percent > minimum_percent * 2 || percent == 100)
+					intensity_desc = ""
+				else if(percent > minimum_percent * 3)
+					intensity_desc = "the strong flavor of"
```

Comment:
Else if never triggers here, percent cannot be more than 3 times than minimum_percent but less than two times at the same time

Reply:
lmfao, didn't notice this when copying it over from the old code
good catch, thanks o7

Reply:
done
![image](https://github.com/user-attachments/assets/d4876147-efe1-4ba2-98cc-701896c9f7f5)
...but actually lemme maybe just like, redo that. entirely. it results in gross sentences like
![image](https://github.com/user-attachments/assets/0ec56f7a-8cd0-416a-87cf-4340b5fca805)